hoomd/state.py

```diff
@@ -198,3 +198,31 @@ def get_group(self, filter_):
             group = _hoomd.ParticleGroup(self._cpp_sys_def, filter_)
             self._groups[cls][filter_] = group
             return group
+
+    def update_group_dof(self):
+        """Update the number of degrees of freedom in each group.
+
+        The groups of particles selected by filters each need to know the number
+        of degrees of freedom given to that group by the simulation's
+        Integrator. This method is called automatically when:
+
+        * The Integrator is attached to the simulation
+
+        Call it manually to force an update.
+        """
+        integrator = self._simulation.operations.integrator
+
+        # Do nothing if there is no integrator.
+        if integrator is None:
+            return
```

Comment:
Assume that `update_group_dof` is the method that later gets triggered when we set a new snapshot. What if the user set's a snapshot before assigning an integrator? Should we have a separate internal method that does nothing when there is no integrator so that the user-facing method can throw an error?